cdap-api/src/main/java/io/cdap/cdap/api/app/ApplicationSpecification.java

```diff
@@ -50,6 +50,11 @@
    */
   String getAppVersion();
 
+  /**
+   * @return CDAP version that was used to create or update pipeline for the Application
+   */
+  String getAppCDAPVersion();
```

Comment:
Is this nullable? What happen for an existing app from an upgrade CDAP?

Reply:
Thank you! I believe I marked it nullable in all other places, will fix here